sklearn/preprocessing/_discretization.py

```diff
@@ -462,13 +462,3 @@ def get_feature_names_out(self, input_features=None):
 
         # ordinal encoding
         return input_features
-
-    def __sklearn_tags__(self):
-        tags = super().__sklearn_tags__()
-        # TODO: fix sample_weight handling of this estimator, see meta-issue #16298
-        tags._xfail_checks = {
-            "check_sample_weight_equivalence": (
-                "sample_weight is not equivalent to removing/repeating samples."
-            ),
-        }
-        return tags
```

Comment:
`KBinsDiscretizer` has actually not problem when the number of samples is small, which is the case in `check_sample_weight_equivalence`.

This check would fail with a larger number of samples and resampling kicks in (and being concurrently investigated in #29907). Or we were to enable `KMeans` which would also make the fit process no longer deterministic even without subsampling. But in either cases, we would need a much slower statistical test instead, so let's better just test the deterministic case for now.